python/cudf/cudf/core/column/categorical.py

```diff
@@ -970,24 +970,30 @@ def find_and_replace(self, to_replace, replacement, all_nan):
         """
         replaced = column.as_column(self.cat().codes)
 
+        to_replace_col, replacement_col = [], []
+        new_cats = cudf.Series(self.dtype.categories)
+        for old_val, new_val in zip(to_replace, replacement):
+            if new_val not in self.dtype.categories:
+                new_cats = new_cats.replace(old_val, new_val)
+            else:
+                to_replace_col.append(self._encode(old_val))
+                replacement_col.append(self._encode(new_val))
+
         to_replace_col = column.as_column(
-            np.asarray(
-                [self._encode(val) for val in to_replace], dtype=replaced.dtype
-            )
+            np.array(to_replace_col, dtype=replaced.dtype)
         )
         replacement_col = column.as_column(
-            np.asarray(
-                [self._encode(val) for val in replacement],
-                dtype=replaced.dtype,
-            )
+            np.array(replacement_col, dtype=replaced.dtype)
```

Comment:
I think this is what I tried initially. The problem I ran into was that we can't currently `_encode` a value that doesn't already exist in as one of the categories in the dtype. So we need to adjust the dtype along the way, and I figured the easiest way of doing that was to just adjust the label that corresponds to those underlying integers, so no actual replacement has to happen.

Reply:
I had to rework this a bit. I hadn't realized that pandas categoricals require the codes to sequentially start from zero with no "gaps". This means we can't have the categories `['one', 'two']` represented by the integers 0 and 2, it has to be 0 and 1. I believe cuDF doesn't actually have the same restriction internally but since we interact with pandas for things like `repr` it will start to throw errors if we aren't compatible.